ios/PurchasesHybridCommon/PurchasesHybridCommonIntegrationTests/StoreKitIntegrationTests.swift

```diff
@@ -116,6 +116,34 @@ class StoreKit1IntegrationTests: BaseIntegrationTests {
         await self.assertSnapshot(result)
     }
 
+    @available(iOS 12.2, macOS 10.14.4, tvOS 12.2, *)
+    func testIneligibleForPromotionalOfferByDefault() async {
+        do {
+            _ = try await CommonFunctionality.promotionalOffer(
+                for: Self.productIdentifier,
+                discountIdentifier: Self.discountIdentifier
+            )
+
+            fail("Expected error")
+        } catch {
+            expect(error).to(matchError(ErrorCode.ineligibleError))
+        }
+    }
+
+    @available(iOS 12.2, macOS 10.14.4, tvOS 12.2, *)
+    func testPromotionalOffer() async throws {
+        try await self.purchaseMonthlyOffering()
+        try self.testSession.expireSubscription(productIdentifier: Self.productIdentifier)
+
+        var result = try await CommonFunctionality.promotionalOffer(
+            for: Self.productIdentifier,
```

Comment:
We could also add a test that checks that it fails getting a promotional offer if passing an incorrect `productIdentifier` or `discountIdentifier`

Reply:
Thinking about it more, it might be too much to try to cover all edge cases in integration tests, so I think it's ok to cover only the major cases like this.

Reply:
It doesn't hurt though, it's trivial. Just added it.